### PR TITLE
adds support for limits on the number of tokens per owner

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- adds support for limits on the number of tokens per owner

## Why are we making these changes?

Allows us to manage the stress we can end placing on our key-value store. In addition, it prevents rogue processes from creating many tokens.

### CODE REVIEW

**Comment by DaoWen** [comment]
I think we should hold off on this until we have good answers to all of these questions:

1. Can users easily obtain a human-readable list of their tokens? This should be human-readable, and for just one account.
2. Do users actually know how to delete tokens?
3. What happens if this limit is hit when token syncing?

**Comment by shamsimam** [comment]
@DaoWen 

> Can users easily obtain a human-readable list of their tokens? This should be human-readable, and for just one account.

This was not previously publicized but can be obtained from the `/tokens?owner-<owner>` endpoint. I have added this entry to our REST api documentation (rest-api.md).

> Do users actually know how to delete tokens?

We have supported token delete and have it documented in our REST api documentation (rest-api.md).

> What happens if this limit is hit during cross-cluster token synchronization?

I have updated the PR to avoid validating the limits during admin mode updates.

**Comment by DaoWen** [comment]
I don't think the "human friendly" aspects of points 1 and 2 in my earlier feedback have been addressed. I think it would be good to wait until David's project is done before merging this. (But we can still review in the meantime.)

File: waiter/src/waiter/token.clj [hunk 1]
```diff
@@ -104,6 +106,21 @@
               owner->owner-key (kv/fetch kv-store token-owners-key)]
           ; Validate the token modification for concurrency races
           (validate-token-modification-based-on-hash existing-token-description version-hash)
+          ; Validate that the maximum number of tokens per owner limit has not been reached
+          (when (integer? token-limit)
```
[the remark this answers is not in the document]
Comment (shamsimam) [reply waiter/src/waiter/token.clj]: Done.

**Pull request merged**